output-validation-rule.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+var assert = require('assert');
+var Joi = require('joi');
+
+module.exports = OutputValidationRule;
+
+function OutputValidationRule(status, spec) {
+  assert(status, 'OutputValidationRule: missing status param');
+  assert(spec, 'OutputValidationRule: missing spec param');
+
+  this.ranges = status.split(',').map(trim).filter(Boolean).map(rangify);
+  this.status = status;
+  this.spec = spec;
+
+  this.validateInput();
+}
+
+/**
+ * Validates it's input values
+ *
+ * @throws Error
+ */
+
+OutputValidationRule.prototype.validateInput = function validateInput() {
+  assert(/^[0-9,\-]+|\*$/.test(this.status),
```

Comment:
Seems like it would be better to make the regex test for groupings of status codes, something like this.

```
/^(?:(?:(?:[2-5][0-9]{2}(?:[,\-]))*?[2-5][0-9]{2})|\*)$/
```

This matches anything from 200-599 for status code, and in any format like so:
```
200
200-399
200,302
400-599
200-299,300-399,400,404
*
```